src/xcode/ENA/ENA/Source/Extensions/FileManager+KeyPackageStorage.swift

```diff
@@ -0,0 +1,51 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+
+/// This extension handles file clean up before new files get written to disk.
+/// New Keyfiles only get written just before a Riskcalculation and since wie have .allreadyRunning this point of time is the safest to remove old keypackages that might gotten left behind.
+/// Keyfiles could be left behind if the App gets killed during the riskcalculation.
+extension FileManager {
+		
+	// MARK: - Internal
+	
+	func createKeyPackageDirectory() throws-> URL {
+		removeOldKeypackages()
+		
+		let url = keyPackageURL().appendingPathComponent(UUID().uuidString, isDirectory: true)
+		try createDirectory(at: url, withIntermediateDirectories: true, attributes: nil)
+		Log.debug("Created: \(url)", log: .localData)
+		return url
+	}
+	
+	// MARK: - Private
+	
+	private func removeOldKeypackages() {
+		Log.debug("Removing \(keyPackageURL())", log: .localData)
+		try? removeItem(at: keyPackageURL()) // Delete the whole dir to make sure no old data is left behind
```

Comment:
Would be nice if we could log if an error happens here.

Reply:
Added.